DuckDuckGo/Freemium/PIR/FreemiumPIRFeature.swift

```diff
@@ -41,18 +43,59 @@ final class DefaultFreemiumPIRFeature: FreemiumPIRFeature {
             4. (Temp) In experiment cohort
          */
         featureFlagger.isFeatureOn(.freemiumPIR) // #1
-        && subscriptionManager.isPrivacyProPurchaseAvailable // #2
-        && !accountManager.isUserAuthenticated // #3
+        && isAPotentialPrivacyProSubscriber // #2 & #3
         // TODO: - Also check experiment cohort here
     }
 
     init(featureFlagger: FeatureFlagger = NSApp.delegateTyped.featureFlagger,
          subscriptionManager: SubscriptionManager,
-         accountManager: AccountManager) {
+         accountManager: AccountManager,
+         freemiumPIRUserStateManager: FreemiumPIRUserStateManager,
+         featureDisabler: DataBrokerProtectionFeatureDisabling = DataBrokerProtectionFeatureDisabler()) {
 
         self.featureFlagger = featureFlagger
         self.subscriptionManager = subscriptionManager
         self.accountManager = accountManager
+        self.freemiumPIRUserStateManager = freemiumPIRUserStateManager
+        self.featureDisabler = featureDisabler
+
+        offBoardIfNecessary()
+    }
+}
+
+private extension DefaultFreemiumPIRFeature {
+
+    /// Returns true if a user is a "potential" Privacy Pro subscriber. This means:
+    ///
+    /// 1. Is eligible to purchase
+    /// 2. Is not a current subscriber
+    var isAPotentialPrivacyProSubscriber: Bool {
```

Comment:
I'd remove the `A` from this name as we don't really use articles in symbol names :)